core/modules/wmgr/python/dbMgr.py

```diff
@@ -0,0 +1,692 @@
+    value = value.strip().lower()
+    if value not in ('0', '1', 'yes', 'no', 'true', 'false'):
+        raise ExceptionResponse(400, "InvalidArgument",
+                                "Unexpected value of '%s' option: \"%s\"" % (option, value))
+    return value in ('1', 'yes', 'true')
+
+def _typeCode2Name(code):
+    """
+    Convert mysql type code to type name, returns None if ther is no mapping.
+    This is mysql-specific, it would be better to have technology-neutral way
+    to provide the same mapping.
+    """
+    for name in dir(FIELD_TYPE):
+        if getattr(FIELD_TYPE, name) == code:
+            return name
+    return None
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+dbService = Blueprint('dbService', __name__, template_folder='dbService')
+
+
+@dbService.errorhandler(db.DbException)
+def dbExceptionHandler(error):
+    """ All leaked DbException exceptions make 500 error """
+    return errorResponse(500, "db.DbException", str(error))
+
+@dbService.route('', methods=['GET'])
+def dbs():
+    """ Return the list of databases """
+
+    _log.debug('request: %s', request)
+    _log.debug('dbs() GET => get database list')
+
+    # use non-privileged account, may limit the list of databases returned.
+    dbConn = Config.instance().dbConn()
+    dbs = dbConn.execCommandN('SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA')
+
+    dbs = [row[0] for row in dbs]
+    _log.debug('dbs(): dbs = %s', dbs)
+
+    dbData = []
+    for dbName in dbs:
+        if dbName not in _specialDbs:
+            dbData += [_dbDict(dbName)]
+
+    return json.jsonify(results=dbData)
+
+
+@dbService.route('', methods=['POST'])
+def dbCreate():
+    """
+    Create new database, database name comes in a query string
+
+    Following parameters are expected to come in a request (in request body
+    with application/x-www-form-urlencoded content like regular form):
+        db: database name (required)
+    """
+
+    _log.debug('request: %s', request)
+    _log.debug('request.form: %s', request.form)
+    _log.debug('dbCreate() POST => make database')
+
+    # get database name from query
+    dbName = request.form.get('db', '').strip()
+    if not dbName:
+        raise ExceptionResponse(400, "MissingArgument", "Database name argument (db) is missing")
+
+    # validate it
+    _validateDbName(dbName)
+
+    # create database, use privileged account
+    dbConn = Config.instance().privDbConn()
+    try:
+        dbConn.createDb(dbName)
+    except db.DbException as exc:
+        _log.error('dbCreate(): exception when creating database %s: %s', dbName, exc)
+        if exc.errCode() == db.DbException.DB_EXISTS:
+            raise ExceptionResponse(409, "DatabaseExists", "Database %s already exists" % dbName)
+        raise
+
+    _log.debug('dbCreate(): database %s created', dbName)
+
+    # grant full access to non-privileged account
+    cmd = "GRANT ALL PRIVILEGES ON {0}.* TO '{1}'@'{2}'"
```

Comment:
Same remark: big security issue if the cluster run in a non-isolated/infiltrated environment. We should provide reasonable security as soon as possible.